sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/WindowMessagesSubscriber.java

```diff
@@ -0,0 +1,927 @@
+        Duration nextMessageTimout, Consumer<T> messageReleaser) {
+        this.loggingContext = Objects.requireNonNull(loggingContext, "'loggingContext' cannot be null.");
+        Objects.requireNonNull(subscriberTerminatedMessage, "'subscriberTerminatedMessage' cannot be null.");
+        this.terminatedMessage = subscriberTerminatedMessage + " (Reason: %s)";
+        this.nextMessageTimout = nextMessageTimout; // can be null.
+        this.messageReleaser = messageReleaser;     // can be null.
+        this.logger = new ClientLogger(WindowMessagesSubscriber.class, loggingContext);
+    }
+
+    /**
+     * Enqueue a request for a window with up to {@code windowSize} number of messages.
+     *
+     * <p>
+     *  TODO (anu) – 1. discuss improving Event Hubs sync receive.
+     *  The WindowMessagesSubscriber can be later moved to azure-core-amqp for both Event Hubs and Service But use.
+     *  For now keeping it in Service Bus to unblock immediate needs.
+     * </p>
+     * <p>
+     *  TODO (anu) – 2. maybe remove 'windowSink' parameter? (discuss with Connie, Liudmila).
+     *  When moving WindowMessagesSubscriber to the azure-core-amqp, do we need to consider passing generic tracer
+     *  interface to trace 'sync receive'?, that enables both Event Hubs and Service Bus tracing, this avoid needing
+     *  call site to pass 'windowSink' parameter. But I think, given the internal usage of this type, the current
+     *  approach of passing 'windowSink' and call site doing trace wiring better than exposing a new generic tracer
+     *  interface impl type.
+     * </p>
+     *
+     * @param windowSize the upper bound for the number of messages to include in the window before closing it.
+     * @param windowTimeout the maximum {@link Duration} since the window was opened before closing it. The window is
+     *     opened when the request gets dequeued to process.
+     * @param windowSink the sink to signal the messages in the window, and to signal termination upon closing window.
+     *
+     * @return the {@link IterableStream} that streams messages and termination signaled to the {@code windowSink}.
+     */
+    public IterableStream<T> enqueueRequest(int windowSize, Duration windowTimeout, Sinks.Many<T> windowSink) {
+        final EnqueueResult<T> r = enqueueRequestIntern(windowSize, windowTimeout, windowSink);
+        return r.getWindowIterable();
+    }
+
+    // Opened in Package-private scope for testing purposes, any library need will use 'enqueueRequest' API.
+    EnqueueResult<T> enqueueRequestIntern(int windowSize, Duration windowTimeout, Sinks.Many<T> windowSink) {
+        if (windowSize <= 0) {
+            throw logger.logExceptionAsError(new IllegalArgumentException("'windowSize' must be strictly positive."));
+        }
+        if (Objects.isNull(windowTimeout) || windowTimeout.isNegative() || windowTimeout.isZero()) {
```

Comment:
I think the Objects.isNull() check doesn't match the error message. I would assume a NullSomethingException would be thrown instead of an IllegalArgument one. See: https://azure.github.io/azure-sdk/java_introduction.html#java-errors-system-errors

Reply:
good catch, updated to use specialized exception.